okhttp/third_party/okhttp/main/java/io/grpc/okhttp/internal/OkHostnameVerifier.java

```diff
@@ -254,4 +258,22 @@ private boolean verifyHostName(String hostName, String pattern) {
     // hostName matches pattern
     return true;
   }
+
+  /**
+   * Normalize the input to lowercase, if it is an ASCII string.
+   * Avoid unicode characters like \u1E0E from returning lowercased ascii
+   * that could match real hostnames.
+   */
+  private static String asciiToLowercase(String input) {
+    if (isAscii(input)) {
+      return input.toLowerCase(Locale.US);
+    } else {
+      return input;
+    }
+  }
+
+  private static boolean isAscii(String input) {
+    // Only ASCII characters are 1 byte in UTF-8.
+    return input.getBytes(StandardCharsets.UTF_8).length == input.length();
```

Comment:
Let's use `com.google.common.base.Utf8.encodedLength()` instead, to avoid creating the `byte[]`.